### ISSUE TITLE
Spectra: Lock "Define View Range"

### ISSUE THREAD

**Issue by Ron024** [Quasars/orange-spectroscopy#365]
It would be nice to be able to lock the view range.  When ever input data is change in the spectra widget it always reverts to auto fit.

**Comment by markotoplak** [comment]
I agree, this would be useful. @borondics, do we have any idea how to clearly visually indicate that a view range is locked?

**Comment by Ron024** [comment]
![image](https://user-images.githubusercontent.com/1922178/62323127-932e4580-b474-11e9-8f53-8892d6515984.png)
Adding a toggle button?  I can try taking a stab at it, but I am not really a GUI guy.

**Comment by Ron024** [comment]
I added a toggle button to the IU.  I don't know how to check it when the screen is updated.
27a40be909dfb647c346575b20008d5703b47465

**Comment by borondics** [comment]
I think this would be a good feature, but maybe the solution can be more generalized. 
We should save the settings in the widget Settings and remember this when a new widget is created. 

To accommodate other datasets and show that there is data outside the view I would recommend the following procedure:
- when data is received plot with saved limits of X, rescale Y to fit everything
- if there are no saved X limits, use autoscale
- if there is data outside the view, put a triangular marker (character ►◀︎ or similar) on the middle of the Y axis in directions where there is additional data
- if there is data, but none of it is in the view, use autoscale

**Comment by borondics** [comment]
@markotoplak For the rescale parameters I had an idea. Instead of keeping it in the Menu, could we pop up a simple two-textbox window when the user double clicks the axis itself that would rescale only the corresponding axis?

It should be pre-filled with the actual limits.
